internal/driver/glfw/canvas.go

```diff
@@ -110,10 +110,15 @@ func (c *glCanvas) SetContent(content fyne.CanvasObject) {
 
 	c.content.Resize(c.content.MinSize()) // give it the space it wants then calculate the real min
 	// the pass above makes some layouts wide enough to wrap, so we ask again what the true min is.
+
 	newSize := c.size.Union(c.canvasSize(c.content.MinSize()))
 	c.Unlock()
 
-	c.Resize(newSize)
+	c.Resize(c.size) // Resize the content to this canvas size
+
+	// If the canvas is to small, we need it to be resized up, but that should be done when the window is actually resized
+	c.context.(*window).Resize(newSize)
+
```

Comment:
I think this is not quite right yet.
First: the canvas should not resize the window IMO. I think the caller of `canvas.SetContent` should handle this.
Second: the canvas should not type-assert its `context` to `*window` just because it is the same by chance. These are two different semantics.
Third: resizing to the current size and then indirectly resizing to the real size really seems to be the wrong thing. There might be something else wrong if this really fixes some misbehaviour.

Reply:
1. I am not sure it is a good idea to move that duty outside of SetContent for a few reason. First one, will be possible forgotten side call when using SetContent (as it would not be a function that work by itself, but require additional step not included). Second is that the logic for resizing the content to any size different from the canvas would have to be moved outside of the canvas logic and I am not sure this belong in window.go.
2. Hum, in that case, what do you recommend? I copied that approach from SetScale. This should also get corrected, I guess.
3. I agree, I just didn't want to undo the first content.Resize due to the comment attached to it and the fear to trigger unwarranted side effect. I feel like removing that double Resize should be part of a different PR later on once this one work properly.

Reply:
Is Window.Resize even needed here? ensureMinSize will maybe take care of it?

Reply:
I don’t have a solution that resolve my objections at hand.
Maybe the whole windows/canvas/content (re)size handling is not right regarding responsibilities?
Probably your changes just revealed some shortcomings.

Reply:
I will try without the Window.Resize here, but indeed the ensureMinSize might do the right work, except with a potential delay. So we might have a frame that get the wrong size that we could have otherwise avoided. Will check.

Reply:
The comment above this line almost explains why it shouldn't exist, not why it does...

Reply:
I guess that my english is not correct then. Could you suggest a better comment in your opinion?

Reply:
Dismissing the stale comments so we can merge